pkg/dataobj/builder.go

```diff
@@ -209,7 +173,7 @@ func (b *Builder) Append(stream logproto.Stream) error {
 	// Since this check only happens after the first call to Append,
 	// b.currentSizeEstimate will always be updated to reflect the size following
 	// the previous append.
-	if b.state != builderStateEmpty && b.currentSizeEstimate+labelsEstimate(ls)+streamSizeEstimate(stream) > int(b.cfg.TargetObjectSize) {
```

Comment:
This was protecting against an edge case where the builder is empty, but a single push payload was far larger than one object can support. This can happen if someone is trying to write really small objects.

If we remove b.state and don't check for empty here, it'll cause an infinite loop where the builder always reports as full but has nothing to flush.